src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -156,56 +159,178 @@ public PaginatedResult getComponents(final Project project, final boolean includ
+                        "B0"."NAME" AS "projectName",
+                        "B0"."PUBLISHER" AS "projectPublisher",
+                        "B0"."PURL" AS "projectPurl",
+                        "B0"."SWIDTAGID" AS "projectSwidTagId",
+                        "B0"."UUID" AS "projectUuid",
+                        "B0"."VERSION" AS "projectVersion",
+                        "A0"."PUBLISHER",
+                        "A0"."PURL",
+                        "A0"."PURLCOORDINATES",
+                        "D0"."ISCUSTOMLICENSE" AS "isCustomLicense",
+                        "D0"."FSFLIBRE" AS "isFsfLibre",
+                        "D0"."LICENSEID" AS "licenseId",
+                        "D0"."ISOSIAPPROVED" AS "isOsiApproved",
+                        "D0"."UUID" AS "licenseUuid",
+                        "A0"."SHA1",
+                        "A0"."SHA_256",
+                        "A0"."SHA_384",
+                        "A0"."SHA3_256",
+                        "A0"."SHA3_384",
+                        "A0"."SHA3_512",
+                        "A0"."SHA_512",
+                        "A0"."SWIDTAGID",
+                        "A0"."UUID",
+                        "A0"."VERSION",
+                        "I0"."LAST_FETCH" AS "lastFetch",
+                        "I0"."PUBLISHED_AT" AS "publishedAt",
+                        "IA"."INTEGRITY_CHECK_STATUS" AS "integrityCheckStatus",
+                        "I0"."REPOSITORY_URL" AS "integrityRepoUrl",
+                        COUNT(*) OVER() AS "totalCount"
+                FROM "COMPONENT" "A0"
+                INNER JOIN "PROJECT" "B0" ON "A0"."PROJECT_ID" = "B0"."ID"
+                LEFT JOIN "INTEGRITY_META_COMPONENT" "I0" ON "A0"."PURL" = "I0"."PURL"
+                LEFT JOIN "INTEGRITY_ANALYSIS" "IA" ON "A0"."ID" = "IA"."COMPONENT_ID"
+                LEFT OUTER JOIN "LICENSE" "D0" ON "A0"."LICENSE_ID" = "D0"."ID"
+                WHERE "A0"."PROJECT_ID" = ?
+                """;
+
         if (onlyOutdated) {
             // Components are considered outdated when metadata does exists, but the version is different than latestVersion
             // Different should always mean version < latestVersion
             // Hack JDO using % instead of .* to get the SQL LIKE clause working:
-            querySring +=
-                    " && !(" +
-                            " SELECT FROM org.dependencytrack.model.RepositoryMetaComponent m " +
-                            " WHERE m.name == this.name " +
-                            " && m.namespace == this.group " +
-                            " && m.latestVersion != this.version " +
-                            " && this.purl.matches('pkg:' + m.repositoryType.toString().toLowerCase() + '/%') " +
-                            " ).isEmpty()";
+            queryString +=
+                    """
+                        AND NOT (NOT EXISTS (
+                        SELECT "M"."ID" AS "DN_APPID"
```

Comment:
`"M"."ID" AS "DN_APPID"` appears to be from the DataNucleus-generated query still, should be safe to remove (or rename, `DN_APPID` doesn't have any meaning).